In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv('survey_data.csv')

In [3]:
df.head()

name   location        job  salary  age  married   gpa
0   Prudy Noelani  Poyntelle      Piano  874297   32    False  2.44
1    Libbi Alisan    Manteno      Proof  193955   51    False  1.09
2  Lyndsay Colley    Raywood      Maker  857501   75    False  1.87
3       Jeni Tine   Rosemead  Solicitor  664514   57     True  1.44
4       Cecil Lon     Drasco      Nanny  875426   49     True  2.02

In [4]:
df.dtypes

name         object
location     object
job          object
salary        int64
age           int64
married        bool
gpa         float64
dtype: object

##### 1.What is the total average Salary?


In [5]:
avg_sal = df['salary'].mean()
print(avg_sal)

509800.014999


**The average salary is 509800.01499**

##### 2. What is the average GPA for people who's last names begin with "A"

In [6]:
# First we need to split the name column into two new column firstname and lastname
df[['firstname','lastname']] = df['name'].str.split(" ",expand=True)


In [7]:
df.head()

name   location        job  salary  age  married   gpa firstname  \
0   Prudy Noelani  Poyntelle      Piano  874297   32    False  2.44     Prudy   
1    Libbi Alisan    Manteno      Proof  193955   51    False  1.09     Libbi   
2  Lyndsay Colley    Raywood      Maker  857501   75    False  1.87   Lyndsay   
3       Jeni Tine   Rosemead  Solicitor  664514   57     True  1.44      Jeni   
4       Cecil Lon     Drasco      Nanny  875426   49     True  2.02     Cecil   

  lastname  
0  Noelani  
1   Alisan  
2   Colley  
3     Tine  
4      Lon

In [8]:
# first we need to select the data whose name start with 'a' in the lastname column so I'm creating a new column whose lastname
#starts with 'A'
avg_gpa = df[df['lastname'].str.startswith('A')]
avg_gpa.head()

name    location      job  salary  age  married   gpa  \
1       Libbi Alisan     Manteno    Proof  193955   51    False  1.09   
29   Kathrine Anjali     Gwinner    Plant  250913   32     True  2.22   
34        Shelbi Axe  Clarksburg  Printer  570498   51     True  1.35   
88    Junette Azalea    Bluffton  Vehicle  275184   71    False  3.16   
131       Floria Asp  Columbiana     Farm   23916   80    False  2.25   

    firstname lastname  
1       Libbi   Alisan  
29   Kathrine   Anjali  
34     Shelbi      Axe  
88    Junette   Azalea  
131    Floria      Asp

In [9]:
avg_gpa['lastname'].count()

70324

##### We have total 70324 people whose last name start with 'A'

In [10]:
# we have got all the data whose name start with 'A' now we can calculate the avg gpa

avg_gpa[['gpa','lastname']].mean()

gpa    2.496632
dtype: float64

**The average gpa of people whose lastname start with 'A' is 2.496632**

##### 3. What is the most popular profession?

In [11]:
df[['job','salary']].max()

job       Zoologist
salary       999998
dtype: object

In [12]:
df.groupby('job')['salary'].max().sort_values(ascending=False).head(1)

job
Nursery    999998
Name: salary, dtype: int64

**As we dont have a complete information on how to select the popular profession so i had selected the popular profession based on the salary. The profession which has the highest salary is taken as popular profession**
**Also i had done groupby on the job column with respective salary column and then the Nursery profession has the maximum salary**

###### 4.What is the median age of people who are married​

In [13]:
df.groupby('married')['age'].median()

married
False    49
True     49
Name: age, dtype: int64

#### The median age of people who are married and not married are same.